src/components/modules/toolbar/blockSettings.ts

```diff
@@ -184,105 +156,44 @@ export default class BlockSettings extends Module<BlockSettingsNodes> {
     }
 
     /** Clear settings */
-    this.nodes.toolSettings.innerHTML = '';
-    this.nodes.defaultSettings.innerHTML = '';
+    if (this.nodes.renderedTunes) {
+      this.nodes.renderedTunes.innerHTML = '';
+    }
 
     /** Tell to subscribers that block settings is closed */
     this.eventsDispatcher.emit(this.events.closed);
 
-    /** Clear cached buttons */
-    this.buttons = [];
-
-    /** Clear focus on active button */
-    this.flipper.deactivate();
+    if (this.popover) {
+      this.popover.off(PopoverEvent.OverlayClicked, this.onOverlayClicked);
+      this.popover.destroy();
+      this.popover.getElement().remove();
+    }
   }
 
   /**
-   * Returns Tools Settings and Default Settings
-   *
-   * @returns {HTMLElement[]}
+   * Adds special class to rendered tool settings signalising that button should
+   * be available for keyboard navigation
    */
-  public get blockTunesButtons(): HTMLElement[] {
+  private makeToolTunesButtonsNavigatable(): void {
     const { StylesAPI } = this.Editor;
-
-    /**
-     * Return from cache
-     * if exists
-     */
-    if (this.buttons.length !== 0) {
-      return this.buttons;
-    }
-
-    const toolSettings = this.nodes.toolSettings.querySelectorAll(
-      // Select buttons and inputs
+    /** Query buttons and inputs inside tunes html */
+    const toolSettings = this.nodes.renderedTunes?.querySelectorAll(
       `.${StylesAPI.classes.settingsButton}, ${$.allInputsSelector}`
     );
-    const defaultSettings = this.nodes.defaultSettings.querySelectorAll(`.${this.CSS.button}`);
-
-    toolSettings.forEach((item) => {
-      this.buttons.push((item as HTMLElement));
-    });
-
-    defaultSettings.forEach((item) => {
-      this.buttons.push((item as HTMLElement));
-    });
 
-    return this.buttons;
-  }
-
-  /**
-   * Add Tool's settings
-   *
-   * @param targetBlock - Block to render settings
-   */
-  private addToolSettings(targetBlock): void {
-    const settingsElement = targetBlock.renderSettings();
-
-    if (settingsElement) {
-      $.append(this.nodes.toolSettings, settingsElement);
+    if (!toolSettings) {
+      return;
     }
-  }
 
-  /**
-   * Add tunes: provided by user and default ones
-   *
-   * @param targetBlock - Block to render its Tunes set
-   */
-  private addTunes(targetBlock): void {
-    const [toolTunes, defaultTunes] = targetBlock.renderTunes();
-
-    $.append(this.nodes.toolSettings, toolTunes);
-    $.append(this.nodes.defaultSettings, defaultTunes);
+    toolSettings.forEach((item) => {
+      item.classList.add(Popover.CSS.itemFlippable);
```

Comment:
That looks strange. Why external dude knows about some internal Popover classname? And then the Popover relies on that external logic inside its internal `flippableElemnts` getter.

If it is a part of the Popover API, it should be presented more explicitly, for example via the public method.

Reply:
logic on determining which elements inside custom content of popover are flippable could be moved to popover itself. But to achieve that StylesApi should be available inside popover. It's easy to pass it as constructor param from blockSettings.ts, but popover is also used in toolbox, where StylesAPI is not available and should be passed from outside as well.

What you see was the easiest solution allowing to prevent passing StylesApi ref through chain of instances. Probably, not the best one